docfx_yaml/extension.py

```diff
@@ -1268,6 +1268,45 @@ def extract_header_from_markdown(mdfile_iterator):
     return mdfile_name
 
 
+# For a given markdown file, adds syntax highlighting to code blocks.
+def highlight_md_codeblocks(mdfile):
+    find_string = '```'
+    find_string_nl = '```\n'
+    replace_string = '```python'
+    new_lines = []
+
+    with open(mdfile) as mdfile_iterator:
+        file_content = mdfile_iterator.read()
+        # If there isn't even number of code block annotations do not syntax
+        # highlight.
+        if file_content.count(find_string_nl) % 2 != 0:
```

Comment:
What happens if a code block actually has a language indicator? For example, this example has valid code fencing, but this check would return an odd number.

````
```python
oops
```
````

Reply:
For what I've seen throughout the libraries this never was the case, nonetheless I should ensure that this case is handled. At the moment we'd return without syntax highlighting applied in the file but this is also not going to work if there's exactly even number of language indicators which would cause chaos.

At the moment the easiest solution should be to check for language indicators' existence, and omit those pair of entries. I'll modify this code a bit to still check that there's an even number of fences.